File: Fibonacci_Number/python/fibonacci_number.py
class Solution:
    def fib(self, n: int) -> int:


        # SIMPLE RECURSION

        # if n == 1:
        #     return 1
        # if n < 1:
        #     return 0
        # return self.fib(n - 1) + self.fib(n - 2)


        #RECURSION WITH MEMOIZATION

        memory = {}
        def fb(num):
            if num == 1:
                return 1
            if num < 1:
                return 0
            if num in memory:
                return memory[num]
            
            memory[num] = fb(num-1) + fb(num-2)
            return memory[num]
        
        return fb(n)

File: Fibonacci_Number/c/fibonacci_number.c
// ITERATIVE Solution

int fib(int n) {
    if (n == 1) {
        return 1;
    }
    else if (n < 1) {
        return 0;
    }
    int a = 0; 
    int b = 1;
    int c = 0;
    for (int i = 2; i < n + 1; i++ ){
        c = a + b;
        a = b;
        b = c;
    }
    return c;
}  

